### PR TITLE
[clkmgr] Fix dft issues

### PR DESCRIPTION
**Pull request by tjaychen**
Addresses #5452 and #5453

- force step_down_req to 0 when in test mode
- remove test mode bypass mux 

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
@arnonsha can you double check the assumptions? I'm not bothering to protect the `step_down` signals after flops since I assume those will be off the scan chain.  If that won't be the case, let me know and I can move the protection point.

**Comment by arnonsha** [comment]
Maybe we should add the  no_scan prefix to all dividers ?
If not we will add the dividers explicitly to the no scan list

**Comment by tjaychen** [comment]
> Maybe we should add the no_scan prefix to all dividers ?
> If not we will add the dividers explicitly to the no scan list

ah yes good point, we had not discussed before if no-scan should have some kind of special naming.  Is no_scan prefix what you prefer? Do you mind adding it to the DFT doc (maybe it's there already and i missed it).

Can I add it to the `prim_clock_div` instantiation level?

**Review by arnonsha — APPROVED** [review]
The no_scan prefix is ok.

**Comment by tjaychen** [comment]
thanks @arnonsha , if someone else can approve, then we can merge this for the next delivery.

File: hw/top_earlgrey/syn/asic.constraints.sdc [hunk 1]
```diff
@@ -70,12 +70,12 @@ set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks IO_CLK]
 
 # generated clocks (div2/div4)
 create_generated_clock -name IO_DIV2_CLK -divide_by 2 \
-    -source [get_pins top_earlgrey/u_clkmgr_aon/u_io_div2_div/u_clk_mux/gen_*u_impl*/u_size_only_mux2/D1] \
-    [get_pins top_earlgrey/u_clkmgr_aon/u_io_div2_div/u_clk_mux/gen_*u_impl*/u_size_only_inv/${DRIVING_CELL_PIN}]
+    -source [get_pins top_earlgrey/u_clkmgr_aon/u_no_scan_io_div2_div/u_clk_mux/gen_*u_impl*/u_size_only_mux2/D1] \
+    [get_pins top_earlgrey/u_clkmgr_aon/u_no_scan_io_div2_div/u_clk_mux/gen_*u_impl*/u_size_only_inv/${DRIVING_CELL_PIN}]
```
Comment (msfschaffner) [inline hw/top_earlgrey/syn/asic.constraints.sdc:74]: These constraints won't work anymore, as they were formulated using the pins of the scan bypass mux cell that is now removed.

Hold on a minute, I need to spin this up in an interactive session to figure out the new constraints.

**Review by msfschaffner — COMMENTED** [review]
Changes look LGTM, but we need to modify the ASIC constraints, see comment below.

File: hw/top_earlgrey/syn/asic.constraints.sdc (hunk 1 above)
Comment (tjaychen) [reply hw/top_earlgrey/syn/asic.constraints.sdc:74]: ah good catch, i didn't notice that.
maybe i can add a buffer in place of the mux?

Comment (tjaychen) [reply hw/top_earlgrey/syn/asic.constraints.sdc]: @msfschaffner can you have a look at the latest?

**Review by msfschaffner — APPROVED** [review]
this LGTM now

**Pull request merged**